pkg/node/heartbeat/client.go

```diff
@@ -0,0 +1,45 @@
+package heartbeat
+
+import (
+	"context"
+
+	natsPubSub "github.com/bacalhau-project/bacalhau/pkg/nats/pubsub"
+	"github.com/bacalhau-project/bacalhau/pkg/pubsub"
+
+	"github.com/nats-io/nats.go"
+)
+
+type HeartbeatClient struct {
+	publisher *natsPubSub.PubSub[Heartbeat]
+	nodeID    string
+}
+
+func NewClient(conn *nats.Conn, nodeID string) (*HeartbeatClient, error) {
+	subParams := natsPubSub.PubSubParams{
+		Subject: heartbeatTopic,
+		Conn:    conn,
+	}
+
+	publisher, err := natsPubSub.NewPubSub[Heartbeat](subParams)
+	if err != nil {
+		return nil, err
+	}
+
+	return &HeartbeatClient{publisher: publisher, nodeID: nodeID}, nil
+}
+
+func (h *HeartbeatClient) Start(ctx context.Context) error {
+	// Waits until we are cancelled and then closes the publisher
+	<-ctx.Done()
+	return h.publisher.Close(ctx)
+}
```

Comment:
`Start` usually implies a non-blocking operation, e.g.  [HeartbeatServer Start method](https://github.com/bacalhau-project/bacalhau/pull/3709/files#diff-901570ddabfcccd3c9bf01ede24a06556fe7cef0d696a10428b74887ba9c7d6bR52). This method is blocking until the context passed to it is canceled. 

I think the change here is to implement this `Start` method in a similar manner to the `HeartbeatServer` stop method - e.g. spawn a go-routine that waits on the context and closes the publisher when its canceled.

Reply:
Actually was able to move this to a Close() method and call it when the containing component shuts down instead. This is the only place that uses it so it feels safe.